src/coreclr/jit/gentree.cpp

```diff
@@ -23130,6 +23132,11 @@ ClassLayout* GenTreeHWIntrinsic::GetLayout(Compiler* compiler) const
 
     switch (GetHWIntrinsicId())
     {
+#ifdef TARGET_XARCH
+        case NI_X86Base_DivRem:
```

Comment:
I think `NI_X86Base_DivRem` should get `typGetBlkLayout(8)`, since it is 8 bytes in size (being a `(uint, uint)` tuple).

Reply:
It can also be `16` in the case of `(nint, nint)`. We need to check the `simdBaseType` and handle it correspondingly -or- ensure that `nint` is always imported as `X64_DivRem` on 64-bit and as `_DivRem` on 32-bit.

Reply:
> Seems like the code already correctly handles this with `genTypeSize(GetSimdBaseType()) * 2`.

Yep.